docs/operations/versioning.md

```diff
@@ -0,0 +1,305 @@
+- Deletion of versions. Before deletion, the operator has to make sure the version is not in use by any Shoot. The recommended way to achieve this, is to deprecate the version and set an expiration date. 
+Clusters with `auto-update` configured will automatically update during their next maintenance. The remainder is forcefully updated during maintenance time after the version is finally expired.
+- Setting an `expiration date` on the latest patch version of an `unmaintained`  Kubernetes minor version. This is to prevent that Gardener automatically forces minor version upgrades for Clusters.
+    - After the deprecation date passed, it will force a Kubernetes minor version update during the next maintenance time window of the Shoot.
+
+## Common scenarios for the Gardener Operator
+
+The following describes common scenarios the Gardener operator regularly faces when managing the Kubernetes and Machine image versions of a Gardener installation.
+We assume a CloudProfile that looks schematically like this:
+
+``` yaml
+1.18.4 :  preview
+1.18.3 : (VersionMaintenance: supported, otherwise: deprecated)
+1.17.0 : (VersionMaintenance: supported, otherwise: deprecated)
+1.16.0 : (VersionMaintenance: supported, otherwise: deprecated)
+1.15.1 : deprecated with no expiration date
+1.15.0 : deprecated with expiration date that is not expired
+1.14.0 : deprecated with expiration date already expired
+```
+For `Version Maintenance`, the Gardener Controller Manager configuration in `controllers.cloudProfile.kubernetesVersionManagement.versionMaintenance` is set to `3`.
+
+**Scenarios**
+
+* * *
+***A new patch version has been released and should be added to the CloudProfile.***
+
+**Example**: Kubernetes releases version 1.18.4. 
+ 
+ -  Add the version and set the classification to `preview`. There can be multiple preview versions.
+ 
+**Example 2**: Kubernetes releases version 1.17.1. 
+ 
+ - With `Version Management` enabled: Version 1.17.1 will be deprecated wit an expiration date right away.
+ 
+ - With `Version Maintenance` enabled: Version 1.17.1 has a maintained minor version. 
+    If it is classified as `preview`, it stays unchanged. If added without a classification, version 1.17.1 will be classified as `supported` and 
+    version 1.17.0 will deprecated with the expiration date for `supported` versions. 
+
+* * *
+***A new minor release version has been released and should be added to the CloudProfile.***
+
+**Example**: Kubernetes releases version 1.19.0.
+
+ - Deprecate the current `supported` version, optionally with an expiration date. Afterwards add the new version as a `preview` version.
+ 
+ - `Version Management` enabled: Add version 1.19.0 without a classification. 1.19.0 will be classified as `supported` and version 1.18.3 will be expired with the configured expiration date.
```

Comment:
Why is 1.19.0 supported right away? Shouldn't it be `preview`?
Also, where is the expiration date for 1.18.3 coming from?

Reply:
1.19.0 can be specified as preview and no automation will touch it. 
These examples describe what the automation does. So adding the version 1.19.0 without a classification triggers the automation (either VersionManagement or VersionMaintenance) to determine an initial classification.

Reply:
The expiration date for 1.18.3 in the case of VersionManagement is configured in the Gardener ControllerManager config [as described above](https://github.com/gardener/gardener/blob/aaca736472c700f9cd9f8b4dbec06b9d2f4f90c0/docs/operations/versioning.md#L138).